pkg/client/snapshot.go

```diff
@@ -0,0 +1,183 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	snapshotv1 "github.com/acorn-io/runtime/pkg/apis/snapshot.storage.k8s.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	snapshot2 "github.com/acorn-io/runtime/pkg/snapshot"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var (
+	annotationAccessModes = "original-pvc-access-modes"
+	annotationResources   = "original-pvc-resource"
+	annotationVolumeName  = "original-pvc-volume-name"
+	annotationVolumeClass = "original-pvc-volume-class"
+)
+
+// SnapshotCreate creates a VolumeSnapshot resource in the cluster using the details of the given *PersistentVolumeClaim.
+// Returns a *VolumeSnapshot or error if one occurred.
+func (c *DefaultClient) SnapshotCreate(ctx context.Context, pvc *corev1.PersistentVolumeClaim) (*snapshotv1.VolumeSnapshot, error) {
+	err := snapshot2.CreateSnapshotClass(ctx, c.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	name, ok := pvc.Labels["custom-name"]
+	if !ok {
+		name = pvc.Labels[labels.AcornPublicName] + "-" + strconv.FormatInt(time.Now().Unix(), 10)
+	}
+
+	accessModesJSON, err := json.Marshal(pvc.Spec.AccessModes)
```

Comment:
I wonder if it is better that you just save the PVC that snapshot is referring to...then you look up the previous PVC to get accessMode and resources..

Reply:
The PVC can be deleted and the snapshot should still be valid

Reply:
yeah, good point 👍